### DISCUSSION TITLE
Packages not found on Visual Studio 2022 with Stable SDK installed

### DISCUSSION THREAD

**Discussion by VArterJr** [microsoft/WindowsAppSDK#1955] [Q&A]
I've tried this on multiple Windows 11 machines, and I can't seem to get it to work. I install the stable SDK from the docs site. That seems to go well. I create a new WinUI C# app in Visual Studio 2022 after that and it creates the project simply fine. But I cannot run, debug, or build the app. I've added nothing so far. Just what the wizard did.

I've tried this on my Surface Laptop, my custom-built desktop (Ryzen 9 based), and various VMs configs. Nothing seems to work, and I just can't find any help online.

Attaching a screenshot of what I see. I'm able to create other C# Windows apps fine (WinForms, UWP). When I try a WinUI app it always just generates a ton of offline package messages (including Microsoft.WIndowsAppSDK, Microsoft.NETCore.app.Runtime.win-x64, etc.).

I pretty much did a full install of VS2022 Enterprise of latest versions of things (newer, not older SDKs and such). Again, I've searched and searched and can't find any help. I've done multiple clean builds of my machines and VMs to test things and nothing seems to help. I've tried ensuring the NuGet packages are all there and OK, and they are.

<img width="1268" alt="Screenshot 2022-01-07 152954" src="https://user-images.githubusercontent.com/4485488/148610035-b02c1bc1-2cae-40a9-9ad3-7b49131e54b4.png">

**Comment by kmgallahan** [comment]
What do things look like under your VS > Options > NuGet Package Manager > General & Package Sources settings?

What does your .csproj look like?

Is logging into Windows and/or VS causing some settings and/or roaming files to be reinstated after the 'clean builds'?

Are you able to automatically download other packages fine or could there be a firewall issue?

Have you tried the single project option? 

https://docs.microsoft.com/en-us/windows/apps/windows-app-sdk/single-project-msix?tabs=csharp

**Comment by VArterJr** [comment]
@kmgallahan , I'm just running the wizard that gets installed as part of the _WindowsAppSDK.Cs.Extension.Dev17.Standalone.vsix_. There doesn't seem to be a version with no install package project included. I can share by project if that would help (GitHub or zip if you prefer). Here is a document of screenshots the wizard and my .csproj file. I'm doing nothing more than running the wizard and trying to build, run, or debug. Nothing else. I've done no other config. Just whatever that wizard does. I'm sure there is some roaming profile going on, but I also tried it in a VM that is not on a corp net. Happens no matter the machine or VM config. Even a bare bones VM with Win 11 and VS 2022.
[App1WinUITest.pdf](https://github.com/microsoft/WindowsAppSDK/files/7836459/App1WinUITest.pdf)

**Comment by zb1749** [comment]
same to me. see in dependencies , you'lll be find a warning that windowsappsdk can't find by default path .  I just try add nuget source, and install windowsappsdk again, every things sees well.

**Comment by riverar** [comment]
Are these machines behind a firewall/proxy/country that could be blocking access to https://api.nuget.org/v3/index.json?

I tested this scenario in a brand new VM and it seems to work fine. I recorded a video that shows start to finish as well. See below.

https://user-images.githubusercontent.com/475132/148750334-0cda4790-55eb-4719-84b1-68d5b447e862.mp4

/rafael

**Reply by VArterJr** [reply]
I figured it out because of another project I had that was missing a dependency. You must add the NuGet 3.x source into VS 2022 manually. Screenshot attached. I had to do this for a third party one too as their "install script" didn't add it automatically. I tried adding the NuGet.org one to my WinUI project and poof. It worked. Don't remember having to do that in 2019 and the prerelease version (Project Reunion). All working beautifully now!

_(@riverar , I checked all the firewall stuff, and it was fine. Nothing blocked. That's how I knew I had to just be missing some setup step.)_

<img width="530" alt="VS2022NuGetSetup" src="https://user-images.githubusercontent.com/4485488/148796855-4f852431-d713-4878-9c0d-85795222edb5.png">

**Reply by VArterJr** [reply]
(I can't mark my own reply as _answer_. Can someone else do that?)

**Reply by riverar** [reply]
That's a default. But I wonder if this setting is synced and you had removed it at some point.

**Reply by VArterJr** [reply]
Don't know. Just know I didn't do it in Project Reunion testing. As far as I remember. ;-)

**Reply by lookinnovative** [reply]
The solution above worked perfectly for me.   nuget.org
 https://api.nuget.org/v3/index.json
